[5.3] Move code to log model in new scheduler history view

laoneo opened the pull request:
### Summary of Changes
This is a followup for #42530 which uses the existing functions of the MVC API and removes some duplicated code. Like that are all ACL checks done, the default events triggered and code looks more inline with the rest of the core.

### Testing Instructions
- Run a task
- Open the new task history view
- Delete a log entry
- Clear the whole history

### Actual result BEFORE applying this Pull Request
All works.

### Expected result AFTER applying this Pull Request
All works.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

laoneo committed:
Move to log model

Kostelano commented on administrator/components/com_scheduler/src/Controller/LogsController.php:
```diff
@@ -73,39 +71,4 @@ public function purge()
-        }
-
-        if (empty($ids)) {
-            $this->setMessage(Text::_('COM_SCHEDULER_NO_LOGS_SELECTED'), 'warning');
```
After deleting the code, the line is no longer used, I suggest deleting it

laoneo replied:
Sure

alikon commented:
just applied the pr and runned a default task

![image](https://github.com/user-attachments/assets/65ff5161-147d-4a8e-b8e6-a6127695a9fd)

laoneo commented:
Please check the response of the ajax request in the browser console. What for an error did you get?

alikon commented:
`Call to undefined method Joomla\CMS\Document\JsonDocument::getToolbar()`

laoneo commented:
Se your server into debug mode and post the whole stack trace please.

alikon commented:
my fault there was an obsolete plugin listening `onBeforeRender`

alikon commented:
I have tested this item :white_check_mark: successfully on 7bbfc2a1c715a6003b7077a42e6929d0f5bba0e4<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44587">issues.joomla.org/tracker/joomla-cms/44587</a>.</sub>

laoneo committed:
remove string

laoneo committed:
types

laoneo committed:
Merge branch '5.3-dev' into scheduler/cleanup

laoneo committed:
Alpha order

exlemor commented:
I have tested this item :white_check_mark: successfully on f9bfbdfbd58ab1974fea5f0d75fb166272a30d72<br /><br />I have successfully tested this as instructed. Worked beautifully.

Thanks laoneo.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44587">issues.joomla.org/tracker/joomla-cms/44587</a>.</sub>

laoneo commented:
As I added only types, removed an obsolete string and alpha ordered the new strings, I mark this as RTC because it needs to go into an alpha.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44587">issues.joomla.org/tracker/joomla-cms/44587</a>.</sub>

laoneo committed:
Merge branch '5.3-dev' into scheduler/cleanup

laoneo committed:
Merge remote-tracking branch 'upstream/5.3-dev' into scheduler/cleanup

rdeutz commented:
Thanks.
